test/integration/array_test.ts

```diff
@@ -444,4 +444,70 @@ describe('Array', function () {
       assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
     }, this.test!.title);
   });
+
+  it('Can handle concurrent insertBefore operations', async function () {
+    await withTwoClientsAndDocuments(async (c1, d1, c2, d2) => {
+      let prev: JSONElement;
+
+      d1.update((root) => {
+        root['k1'] = [1];
+        prev = root['k1'].getElementByIndex(0);
+      });
+      await c1.sync();
+      await c2.sync();
+      assert.equal(d1.toSortedJSON(), d2.toSortedJSON());
+
+      d1.update((root) => {
+        root['k1'].deleteByID(prev.getID());
+        assert.equal('{"k1":[]}', root.toJSON());
+      });
+      d2.update((root) => {
+        root['k1'].insertBefore(prev.getID(), 2);
+        assert.equal('{"k1":[2,1]}', root.toJSON());
+      });
```

Comment:
It is not normal to insert a new element based on the deleted element.
It looks good to treat as an error, what do you think?

Reply:
What do you think of this situation, @mojosoeun?

Reply:
I'd like to register as a new issue.

Reply:
Sorry, I didn't catch that d1 and d2 are not synchronized. We can ignore this comment.